js/tests/unit/collapse.js

```diff
@@ -177,6 +177,34 @@ $(function () {
     setTimeout(done, 0)
   })
 
+  QUnit.test('should not show a collapse when initialized with "hide" if already hidden', function (assert) {
+    assert.expect(0)
+    var done = assert.async()
+
+    $('<div class="collapse"></div>')
+      .appendTo('#qunit-fixture')
+      .on('show.bs.collapse', function () {
+        assert.ok(false, 'showing a previously-uninitialized hidden collapse when "hide" method called')
+      })
+      .bootstrapCollapse('hide')
+
+    setTimeout(done, 0)
+  })
+
+  QUnit.test('should hide a collapse when initialized with "hide" if not already hidden', function (assert) {
+    assert.expect(1)
+    var done = assert.async()
+
+    $('<div class="collapse in"></div>')
```

Comment:
That's not valid HTML and such shortcuts have caused some problems in the past (I think it with IE8, but still).